### DISCUSSION TITLE
Javascript error - cannot set property 'display' of undefined

### DISCUSSION THREAD

**Discussion by igorfuna** [ajaxorg/ace#4807] [General]
I'm using ace in a wordpress plugin. It works fine on 200 000+ installations.
However, on one site it does not work - it reports Javascript error - cannot set property 'height' of undefined.

Tested with 1.4.5 - the last line in this code throws the error:

```
var Lines = function(element, canvasHeight) {
    this.element = element;
    this.canvasHeight = canvasHeight || 500000;
    this.element.style.height = (this.canvasHeight * 2) + "px";
```
This is where the Lines function is called:

```
var Gutter = function(parentEl) {
    this.element = dom.createElement("div");
    this.element.className = "ace_layer ace_gutter-layer";
    parentEl.appendChild(this.element);
    this.setShowFoldWidgets(this.$showFoldWidgets);
    this.gutterWidth = 0;
    this.$annotations = [];
    this.$updateAnnotations = this.$updateAnnotations.bind(this);
    this.$lines = new Lines(this.element);
    this.$lines.$offsetCoefficient = 1;
};
```

This can be reproduced only on one website (specific combination of WP theme and plugins). 
Any clue?

**Comment by nightwing** [comment]
So the element is defined but doesn't have style? Maybe some code is changing `document.createElementNS` could you try to check the value of `this.element` in the debugger?

**Comment by igorfuna** [comment]
`<div class="ace_layer ace_gutter-layer ace_folding-enabled"></div>`

![js](https://user-images.githubusercontent.com/7113729/63799986-7bc86800-c90d-11e9-984c-22784e090500.png)

**Comment by eladavron** [comment]
I'm encountering a weirdly similar issue.
The same line as in @igorfuna 's message throws this exception:
```js
TypeError: Cannot set property 'height' of undefined
    at new s (239.js:formatted:11691)
    at new l (239.js:formatted:11797)
    at new w (239.js:formatted:13143)
    at Module.ace.define.t.edit (239.js:formatted:15980)
    at main.js:formatted:72556
    at fi (main.js:formatted:59833)
    at t.unstable_runWithPriority (main.js:formatted:64344)
    at Wa (main.js:formatted:56490)
    at zi (main.js:formatted:59622)
    at ri (main.js:formatted:59362)
to @ main.js:formatted:58633
```

But weirdly enough - **only when hosted by Flask!**
When hosted by the Webpack dev server it works just fine!

I put a breakpoint there, and when in Flask, the object: `this.element` has no `style` property, but when in Webpack it shows this for `this.element.style`: 

```js
CSSStyleDeclaration {alignContent: "", alignItems: "", alignSelf: "", alignmentBaseline: "", all: "", …}
```

If you have any idea what's causing this or how to overcome it I would be greatly appreciative.
